src/plugins/dashboard/public/application/legacy_app.ts

```diff
@@ -127,14 +147,38 @@ export function initDashboardApp(app, deps) {
           $scope.create = () => {
             history.push(DashboardConstants.CREATE_NEW_DASHBOARD_URL);
           };
+          $scope.editItemAvailable = (item) => !!item.editUrl;
           $scope.find = (search) => {
-            return service.find(search, $scope.listingLimit);
+            const dashboardList$ = from(service.find(search, $scope.listingLimit));
+            const dashboardListItems$ = dashboardList$.pipe(
+              mergeMap((l) => l.hits),
+              map((item) => ({
+                ...item,
+                editUrl: `${createDashboardEditUrl(item.id)}?_a=(viewMode:edit)`,
+                type: 'Dashboard',
+                listingType: 'dashboard',
+              }))
+            );
+            const otherDashboardLists$ = from(deps.dashboardListSources).pipe(
+              mergeMap(({ name, listProviderFn }) => listProviderFn()) // execute each list source
+            );
+
+            const combined$ = concat(dashboardListItems$, otherDashboardLists$).pipe(
+              reduce<DashboardListItem, DashboardDisplay>(
+                (acc: DashboardDisplay, item: DashboardListItem) => {
+                  return { hits: [...acc.hits, item], total: acc.total + 1 };
+                },
+                { hits: [], total: 0 }
+              )
+            );
+
+            return combined$.toPromise();
           };
-          $scope.editItem = ({ id }) => {
-            history.push(`${createDashboardEditUrl(id)}?_a=(viewMode:edit)`);
+          $scope.editItem = ({ editUrl }) => {
+            history.push(editUrl);
           };
-          $scope.getViewUrl = ({ id }) => {
-            return deps.addBasePath(`#${createDashboardEditUrl(id)}`);
+          $scope.getViewUrl = ({ url }) => {
+            return url;
```

Comment:
Does this work for the default dashboard URL's too? since it looks like you are no longer performing this 
```
deps.addBasePath(`#${createDashboardEditUrl(id)}`)
```

Reply:
Yes, still works.  React router assumes base-path.